src/bulk-load.js

```diff
@@ -129,12 +129,15 @@ module.exports = class BulkLoad extends EventEmitter {
     const arr = row instanceof Array;
     for (let i = 0, len = this.columns.length; i < len; i++) {
       const c = this.columns[i];
-      c.type.writeParameterData(this.rowsData, {
+
+      const parameter = {
         length: c.length,
         scale: c.scale,
         precision: c.precision,
-        value: row[arr ? i : c.objName]
-      }, this.options);
+        value: c.type.validate(row[arr ? i : c.objName], c.length, c.precision, c.scale)
```

Comment:
`validate()` doesn't throw error but it returns error if validation fails. Can you add check to verify type of  `validate()`'s return value before passing it to `writeParameterData`?